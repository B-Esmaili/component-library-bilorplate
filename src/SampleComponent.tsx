import * as React from 'react';
import {Card} from 'reactstrap';

interface Props {
  className?: string;
  title: string;
  description: string;
}

export const SampleComponent: React.FunctionComponent<Props> = ({
  className,
  title,
  description,
}) => (
  <Card className={className}>
    <h1>{title}</h1>
    <p>{description}</p>
  </Card>
);
