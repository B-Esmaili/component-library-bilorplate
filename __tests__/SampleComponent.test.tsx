import {shallow} from 'enzyme';
import * as React from 'react';

import {SampleComponent} from '../src/SampleComponent';

const hydratedProps = {
  className: 'p-2 m-2',
  title: 'title',
  description: 'description',
};

describe('<SampleComponent />', () => {
  it('should render and match snapshot', () => {
    const tree = shallow(<SampleComponent {...hydratedProps} />);

    expect(tree).toMatchSnapshot();
  });
});
