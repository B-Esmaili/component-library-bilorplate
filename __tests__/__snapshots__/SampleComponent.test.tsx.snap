// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleComponent /> should render and match snapshot 1`] = `
<Card
  className="p-2 m-2"
  tag="div"
>
  <h1>
    title
  </h1>
  <p>
    description
  </p>
</Card>
`;
